src/Tizen.NUI.WindowSystem/src/public/InputGesture.cs

```diff
@@ -0,0 +1,557 @@
+        Left,
+    }
+
+    /// <summary>
+    /// Enumeration of gesture edge sizes.
+    /// </summary>
+    /// This enum is need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public enum GestureEdgeSize
+    {
+        /// <summary>
+        /// edge size none.
+        /// </summary>
+        None,
+
+        /// <summary>
+        /// edge size full.
+        /// </summary>
+        Full,
+
+        /// <summary>
+        /// edge size partial.
+        /// </summary>
+        Partial,
+    }
+
+    /// <summary>
+    /// Class for the Tizen Input Gesture.
+    /// </summary>
+    /// <privilege>
+    /// http://tizen.org/privilege/gesturegrab
+    /// </privilege>
+    /// This class is need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class InputGesture : IDisposable
+    {
+        private IntPtr _handler;
+        private bool disposed = false;
+        private bool isDisposeQueued = false;
+
+        private event EventHandler<EdgeSwipeEventArgs> _edgeSwipeEventHandler;
+        private EdgeSwipeCb _edgeSwipeDelegate;
+
+        private event EventHandler<EdgeDragEventArgs> _edgeDragEventHandler;
+        private EdgeDragCb _edgeDragDelegate;
+
+        private event EventHandler<TapEventArgs> _tapEventHandler;
+        private TapCb _tapDelegate;
+
+        private event EventHandler<PalmCoverEventArgs> _palmCoverEventHandler;
+        private PalmCoverCb _palmCoverDelegate;
+
+        internal void ErrorCodeThrow(Interop.InputGesture.ErrorCode error)
+        {
+            switch (error)
+            {
+                case Interop.InputGesture.ErrorCode.None :
+                    return;
+                case Interop.InputGesture.ErrorCode.OutOfMemory :
+                    throw new Tizen.Applications.Exceptions.OutOfMemoryException("Out of Memory");
+                case Interop.InputGesture.ErrorCode.InvalidParameter :
+                    throw new ArgumentException("Invalid Parameter");
+                case Interop.InputGesture.ErrorCode.PermissionDenied :
+                    throw new Tizen.Applications.Exceptions.PermissionDeniedException("Permission denied");
+                case Interop.InputGesture.ErrorCode.NotSupported :
+                    throw new NotSupportedException("Not Supported");
+                default :
+                    throw new InvalidOperationException("Unknown Error");
+            }
+        }
+
+        /// <summary>
+        /// Creates a new InputGesture.
+        /// </summary>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="Tizen.Applications.Exceptions.PermissionDeniedException">Thrown when the permission is denied.</exception>
+        public InputGesture()
```

Comment:
Since this module is using the dispose queue from NUI and binding EFL, it would be good to add comments like this: This module operates in a NUI application and requires instantiation and disposal on the main thread. 
but I'm simply stating my own opinion here, so you can consider it.